src/monitor/monitor.c

```diff
@@ -1642,6 +1642,12 @@ errno_t load_configuration(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
+    /* return EOK so genconf=KCM may start without any
+     * sssd configuration */
+    if (only_section != NULL) {
+        goto done;
```

Comment:
> Do we really want to always skip following steps (even if config exist)?

IIUC the remaining function calls in load_configuration ``get_monitor_config()`` and ``chown()`` will not do anything when ``--genconf_section`` is used.

> 
> In particular, `ret == EOK` => `ctx` isn't freed, but it's not stored in `*monitor` either. Isn't this a mem leak? But probably doesn't matter since process will exit right after this call..?
> 
> (I didn't read this code carefully, so might ask silly questions, sorry)

Yes I see this in valgrind, but the process exits after load_configuratoins returns, is it worth addressing still?

Reply:
Hi,

although not strictly needed I think it would be more clean to free it and assign NULL to *monitor before leaving load_configuration().

bye,
Sumit

Reply:
Updated, please check.

Reply:
Thank you.